lib/delta_tables_generator/importer.rb

```diff
@@ -0,0 +1,191 @@
+module DeltaTablesGenerator
+  class Importer
+    class << self
+      DB = Sequel::Model.db
+
+      def label
+        'item'
+      end
+
+      def perform_import(day: Date.current)
+        raise NotImplementedError, 'Implement this method in the subclasses'
+      end
+
+      def perform_backlog_import(from: Date.current - 3.months, to: Date.current)
+        (from..to).each do |day|
+          perform_import(day: day)
+        end
+      end
+
+      def table
+        :goods_nomenclatures
+      end
+
+      def import_fields
+        %i[goods_nomenclature_item_id
+           goods_nomenclature_sid
+           productline_suffix
+           end_line
+           delta_type
+           delta_date]
+      end
+
+      def where_condition(day: Date.current)
+        'true'
```

Comment:
Should this not be `raise NotImplementedError, 'Implement this method in the subclasses' ? Looking at the children this method is supposed the return a SQL statement as string.

Reply:
Good point, thanks!